LwM2M Client LTE network registration status notify and automatic LwM2M engine pause/resume

jheiskan81 opened the pull request:
Extended neighbor cell notification module to support also LTE network registration status notification.
LwM2M client sample enable LTE network status monitoring and automatically pause and resume LwM2M engine based on client connected status and LTE network status.

de-nordic committed:
net/lib/nrf_cloud: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macro.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net/lwm2m_client_utils: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net: lib: azure_iot_hub: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
app: serial_lte_modem: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
applications: nrf_desktop: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
applications: asset_tracker_v2: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
applications: connectivity_bridge: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
lib: location: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
lib: lte_link_control: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
lib: multicell_location: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
lib: sms: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
bluetooth: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>
Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

de-nordic committed:
caf: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
test: bluetooth: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
nrf_rpc: bluetooth: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
nfc: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net: lib: rest_client: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net: lib: zzhc: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
lib: ftp_client: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
fota_download: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net: lib: download_client: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net: lib: azure_fota: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net: lib: aws_jobs: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net: lib: aws_iot: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
net: lib: aws_fota: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
mpsl: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
event_manager_proxy: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
dfu_target: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
event_manager: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
samples/zigbee: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
samples/peripheral/802154_phy_test: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
samples: openthread: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
samples: nrf9160: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
modules: memfault: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
samples: bluetooth: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
lib: at_monitor: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
lib: nrf_modem: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
lib: modem_info: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
partition_manager: Fix mergehex.py path

The file has been moved.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
partition_manager: Remove usage of fa_dev_name

Zephyr has removed fa_dev_name from the flash_area structure;
this commit removes usage of fa_dev_name from partition manager
and replaces it, for the purpose of device identification,
with usage of DTS node labels.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
mcuboot: nrf53_hooks: Replace device_get_binding

THe DEVICE_DT_GET is used instead as we no longer reference device
by name and use DTS node label instead.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
subsys/dfu_target: Move dfu_target_mcuboot to PM without _DEV_NAME

The commit modifies source code to remove usage of PM*_DEV_NAME
defines, which are no longer provided by the partition manager.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
fprotect: Move from using PM_*_DEV_NAME to access devices

The commit replaces usage of PM_*_DEV_NAME with PM_*_DEV, as the
former is no longer provided by the partition manager.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
application/serial_lte_modem: Move from using PM_*_DEV_NAME

The commit replaces usage of PM_*_DEV_NAME, which is no longer
provided by the partition manager, with PM_*_DEV.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
fw_info: Move from using PM_*_DEV_NAME to PM_*_DEV

The partition manager no longer provides PM_*_DEV_NAME definitions
and any references to these have to be replaced with PM_*_DEV,
accompanied by a supporting code.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nordicjm committed:
manifest: Update west.yml with upmerge changes

Updates west.yml with upmerge changes as of 4th August 2022

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
manifest: Move to zephyr zcbor repository in west.yml

Use the zephyr zcbor repository instead of a separate Nordic one

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
dm: Update paths to include zephyr prefix

The zephyr prefix is needed to include these files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
emds: Update paths to include zephyr prefix

The zephyr prefix is needed to include these files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
net: lib: nrf_cloud: Update paths to include zephyr prefix

The zephyr prefix is needed to include these files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: peripheral_fast_pair: Update paths to include zephyr prefix

The zephyr prefix is needed to include these files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: matter: Update paths to include zephyr prefix

The zephyr prefix is needed to include these files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: lpuart: Update paths to include zephyr prefix

The zephyr prefix is needed to include these files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
nrf_rpc: Update paths to include zephyr prefix

The zephyr prefix is needed to include these files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: event_manager_proxy: Update paths to include zephyr prefix

The zephyr prefix is needed to include these files

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: psa_tls: Update log Kconfig

LOG2 prefix is now LOG as v2 logging is the default

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

de-nordic committed:
bootloader: The partition manager no longer uses labels

The partition manager has been moved from using label property,
to identify device, to use DTS node or DTS nodelabel.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nordicjm committed:
zigbee: lib: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
net: lib: lwm2m: Remove log_strdup invocations

It is no longer needed to call log_strdup while providing string
for LOG_ macros.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
Bluetooth: tester: Fix uart_pipe path

The location of the uart_pipe.h file was moved up a directory

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
lib: nrf_modem_lib: Update log functions

Logging v2 has became the default and the 2 is no longer needed

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
drivers: serial: lpuart: Switch to updated device tree macros

Uses DEVICE_DT_GET() macros instead of device_get_binding()

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
drivers: wifi: nrf700x: Fix GPIO flag

Fixes the GPIO flag to use the Nordic specific flag

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: ipc: Add ifdef protection for test extra stack size

Prevents using test extra stack size if it is not defined

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

kapi-no committed:
bluetooth: services: ancs_client: fix warning for uninitialized struct

Fixed a build warning in the Peripheral ANCS Client sample that was
caused by uninitialized notification structure in the ANCS client
module.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

nordicjm committed:
samples: debug: ppi_trace: Add nrf52840dk_nrf52840 overlay for entropy

An overlay is needed to set the default entropy source to the nRF5 RNG
instead of the cryptocell

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: lib: nrf_modem: trace: Switch to pinctrl for qemu m3 target

This switches to pinctrl for the UART trace nrf_modem test

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: tfm_secure_peripheral: Add pinctrl SPI definitions

Prevents build failures in the sample application by moving to pinctrl

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

Commit:
drivers: mpsl: fix TEMP_NRF5_MPSL option

With
https://github.com/zephyrproject-rtos/zephyr/commit/f0441fba58a242d4fcc70794f6250ace358baae4
merged upstream, `TEMP_NRF5` is enabled by default if temperature sensor
is defined in devicetree. That caused build errors, as `TEMP_NRF5_MPSL`
option required `TEMP_NRF5` to be disabled.

This commit fixes that error by treating `TEMP_NRF5_MPSL` as an
alternative implementation of temperature driver, selected through
`TEMP_NRF5_FORCE_ALT`. The intended usage of that symbol is to allow the
application to select `TEMP_NRF5`, but to use a different underlying
source code for actually driving the sensor. `TEMP_NRF5_MPSL` is aligned
to achieve that.

Signed-off-by: Jedrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

nordicjm committed:
emds: Use device struct instead of device name for flash device

Updates the flash device to use the device structure directly instead
of the flash area name, which is no longer supported

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
emds: Add zephyr prefix

Adds the zephyr/ prefix when including files from Zephyr

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: emds: emds_flash: Use device struct instead of flash area name

Fixes a build issue due to the device name no longer being available
and uses the flash device directly

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: nrf5340_audio_dk_nrf5340: Remove USB labels

USB labels are no longer supported, this remove the headset labels

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
dfu: dfu_target: Fix mcuboot parameters from partition manager

Fixes extracting information from partition manager for mcuboot DFU
target

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

de-nordic committed:
partition_manager: Workaround for building with no flash enabled

The commit adds workaround for partition map failing, while doing
multi-image build, for sub-images that do not use or enable flash,
but enable PM build through inheritance of PM configuration from
main image.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nordicjm committed:
partition_manager: Add missing includes

Add includes needed to set up an empty flash map

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: dfu: dfu_target_mcuboot: Update static PM file

Adds updated configuration values to the static partition manager
test file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
flash_map: pm: Update to use device instead of device name

Updates the partition manager configuration to use the device directly
rather than by using the device name which is no longer supported

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
mpsl: fem: Move from GPIO port labels to GPIO port paths

Labels are no longer supported, this updates the FEM GPIO labels to
use port paths instead

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

eivindj-nordic committed:
lib: at_host: use uart device node label instead of device name

Updates the at_host uart to use the device node label instead of
the device name which is no longer supported.

Signed-off-by: Eivind Jølsgard <eivind.jolsgard@nordicsemi.no>

de-nordic committed:
applications/machine_learning: Set UART forwarder via DTS overlay

With removal of label properties, from DTS tree, it is no longer
possible to identify UART by string label. This also means that
it is no longer possible to use device_get_binding to obtain
pointer to UART device, and DEVICE_DT_GET needs to be used instead.
Unfortunately this makes Kconfig option
CONFIG_ML_APP_EI_DATA_FORWARDER_UART_DEV obsolete and DTS overaly
is now needed to attach ml_app_ei_data_forwarder_uart DTS node
label to UART that will be used for data forwarder.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nordicjm committed:
applications: nrf5340_audio: Remove old parameter from function call

The extra parameter is no longer part of the function definition

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
applications: nrf5340_audio: Remove mics.h include

This file no longer exists and is not needed

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

MarekPieta committed:
applications: machine_learning: Remove CONFIG_BT_HCI_ECC_STACK_SIZE

The CONFIG_BT_HCI_ECC_STACK_SIZE was removed from Zephyr.

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
doc: known_issues: Remove link from CONFIG_BT_HCI_ECC_STACK_SIZE

The CONFIG_BT_HCI_ECC_STACK_SIZE was removed from Zephyr, so the
link would not work.

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

Commit:
samples: lwm2m_client: Fix missing event handler

Added event handler for LWM2M_RD_CLIENT_EVENT_ENGINE_SUSPENDED.

Signed-off-by: Juha Heiskanen <juha.heiskanen@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Disable boot banner

Change disables boot banner to make configurations more consistent.
The change is also needed to make MCUboot fit on nrf52820dongle.

Jira: NCSDK-16346

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Reduce FLASH usage on nrf52820 boards

Change reduces FLASH usage of MCUboot on nrf52820 boards.

Jira: NCSDK-16346

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

Commit:
mpsl: fem: remove deprecated DT_GPIO_LABEL from simple_gpio

This commit removes the usage of deprecated DT_GPIO_LABEL macro from
mpsl_fem_simple_gpio implementation.

Signed-off-by: Jedrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

Commit:
samples: lwm2m_client: LwM2M pause and resume automatically

Use new LTE network registration notification for pause and resume
LwM2M engine.

Signed-off-by: Juha Heiskanen <juha.heiskanen@nordicsemi.no>

Commit:
net: lib: lwm2m_client_utils: LTE network state notification

Extented neighbour cell notification module for support also
network registration update notification.

Signed-off-by: Juha Heiskanen <juha.heiskanen@nordicsemi.no>

jheiskan81 commented:
@rlubos @SeppoTakalo @jarlamsa @VeijoPesonen Please review.

jheiskan81 commented:
@nordicjm Do not merge untill review is ready. If Upmerge is ready before that I do normal PR to nrf-sdk.

VeijoPesonen commented on samples/nrf9160/lwm2m_client/src/main.c:
```diff
@@ -382,6 +392,29 @@ static void modem_connect(void)
+			}
+		}
+	} else {
+		LOG_INF("LTE disconne disconnected");
```
```suggestion
		LOG_INF("LTE disconnected");
```

jarlamsa replied:
Nitpick: LTE disconnected

VeijoPesonen commented on subsys/net/lib/lwm2m_client_utils/Kconfig:
```diff
@@ -132,6 +132,9 @@ config LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS_BUF_SIZE
 	default 4096
 
+config LWM2M_CLIENT_UTILS_NWK_REG_NOTIFICATION
+	bool "Enable LE Network registration status Notification"
```
```suggestion
	bool "Enable LTE Network registration status Notification"
```

jarlamsa commented on include/net/lwm2m_client_utils.h:
```diff
@@ -172,6 +172,26 @@ void location_assist_cell_request_set(void);
+typedef void (*lwm2m_lte_nwk_reg_update_state_cb_t)(bool connected);
+
+/**
+ * @brief LTE network registration update event order.
```
Suggestion: Register a callback for receiving LTE network status notifications.

jarlamsa commented on include/net/lwm2m_client_utils.h:
```diff
@@ -172,6 +172,26 @@ void location_assist_cell_request_set(void);
+/**
+ * @brief LTE network registration update event order.
+ *
+ * @param[in] cb A callback function to receive lte network state changes
```
Suggestion: cb A callback function to receive LTE network status notifications.

jarlamsa commented on samples/nrf9160/lwm2m_client/src/main.c:
```diff
@@ -335,19 +344,20 @@ static void rd_client_event(struct lwm2m_ctx *client, enum lwm2m_rd_client_event
 	case LWM2M_RD_CLIENT_EVENT_QUEUE_MODE_RX_OFF:
 		LOG_DBG("Queue mode RX window closed");
 		break;
-
```
Nitpick: Don't remove empty line here.

VeijoPesonen commented on samples/nrf9160/lwm2m_client/src/main.c:
```diff
@@ -382,6 +392,29 @@ static void modem_connect(void)
 	} while (ret < 0);
 }
 
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_SIGNAL_MEAS_INFO_OBJ_SUPPORT)
```
This seems to be wrong.

jarlamsa commented on samples/nrf9160/lwm2m_client/src/main.c:
```diff
@@ -382,6 +392,29 @@ static void modem_connect(void)
+	} else {
+		LOG_INF("LTE disconne disconnected");
+		if (!lwm2m_stack_suspended && lwm2m_client_connected) {
+			/* Are we connected already */
```
This is a bit confusing comment, consider removing.

VeijoPesonen commented on samples/nrf9160/lwm2m_client/src/main.c:
```diff
@@ -429,6 +462,10 @@ void main(void)
 		return;
 	}
 
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_SIGNAL_MEAS_INFO_OBJ_SUPPORT)
```
This seems to be wrong too.

jarlamsa commented on samples/nrf9160/lwm2m_client/src/main.c:
```diff
@@ -60,6 +60,11 @@ static int reconnection_counter;
 static bool update_session_lifetime = true;
+static bool lwm2m_client_connected;
+
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_SIGNAL_MEAS_INFO_OBJ_SUPPORT)
```
Why the suspend is now depending of the signal measurement info object that is used just for neighbor cell measurements?

jheiskan81 replied:
I fix this already

jarlamsa commented on subsys/net/lib/lwm2m_client_utils/Kconfig:
```diff
@@ -132,6 +132,9 @@ config LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS_BUF_SIZE
 	default 4096
 
+config LWM2M_CLIENT_UTILS_NWK_REG_NOTIFICATION
+	bool "Enable LE Network registration status Notification"
```
Suggestion: "Enable LTE network registration status notifications"

jheiskan81 replied:
fixed.

jheiskan81 commented:
@VeijoPesonen Thanks for good findings I have been fixed those.

jarlamsa commented on subsys/net/lib/lwm2m_client_utils/CMakeLists.txt:
```diff
@@ -14,8 +14,8 @@ zephyr_library_sources_ifdef(CONFIG_LWM2M_CLIENT_UTILS_DEVICE_OBJ_SUPPORT lwm2m/
-zephyr_library_sources_ifdef(CONFIG_LWM2M_CLIENT_UTILS_NEIGHBOUR_CELL_LISTENER
-			     lwm2m/neighbour_cell_info.c)
+zephyr_library_sources_ifdef(CONFIG_LWM2M_CLIENT_UTILS
+			     lwm2m/lwm2m_lte_notification.c)
```
Changing this dependency causes some issues as the `CONFIG_LWM2M_CLIENT_UTILS_NEIGHBOUR_CELL_LISTENER` is dependent of `CONFIG_LWM2M_CLIENT_UTILS_SIGNAL_MEAS_INFO_OBJ_SUPPORT`
The code shouldn't try to populate the neighbour cell information to object if the object itself isn't supported.

jheiskan81 replied:
good point I change lwm2m_lte_notification() to use CONFIG_LWM2M_CLIENT_UTILS_NEIGHBOUR_CELL_LISTENER

jarlamsa commented on subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_lte_notification.c:
```diff
@@ -94,9 +98,57 @@ int lwm2m_update_signal_meas_objects(const struct lte_lc_cells_info *const cells
+	return 0;
+}
+
+static void lwm2m_lte_reg_handler_notifify(enum lte_lc_nw_reg_status nw_reg_status)
```
nitpick: notify

jarlamsa reviewed (changes requested):
The kconfig dependencies now cause issues. The network status notifications shouldn't be dependent of the signal measurement info objects nor should the measurement info objects be used when they aren't configured to be used.

jheiskan81 commented:
@jarlamsa Thanks for valuable feedback. I have fixed comment's and findings.

jarlamsa commented on subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_lte_notification.c:
```diff
@@ -17,6 +17,10 @@
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(lwm2m_neighbour_cell, CONFIG_LWM2M_CLIENT_UTILS_LOG_LEVEL);
```
Maybe change the module name as the module isn't just about ncell anymore.

jheiskan81 replied:
true I can do it.

jarlamsa commented on subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_lte_notification.c:
```diff
@@ -17,6 +17,10 @@
 LOG_MODULE_REGISTER(lwm2m_neighbour_cell, CONFIG_LWM2M_CLIENT_UTILS_LOG_LEVEL);
+static void lte_notify_handler(const struct lte_lc_evt *const evt);
+
+static bool event_handler_requested;
```
These 2 should be in the `#ifdef(CONFIG_LWM2M_CLIENT_UTILS_NWK_REG_NOTIFICATION)`

jheiskan81 replied:
No these 2 are needed for both ncell and Network registration notify.

jarlamsa replied:
Oh yeah, true.

jarlamsa reviewed (changes requested):
2 small things only, otherwise looking good already.

nordicjm committed:
mpsl: fem: Replace label comparison with device tree macros

This tidies up the MPSL FEM GPIO configuration code to use device tree
macros rather than checking the string of the GPIO port to detemine a
pin mapping

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

jarlamsa commented on subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_lte_notification.c:
```diff
@@ -94,9 +98,57 @@ int lwm2m_update_signal_meas_objects(const struct lte_lc_cells_info *const cells
+{
+if (!event_handler_requested) {
+		event_handler_requested = true;
+		lte_lc_register_handler(lte_notify_handler);
```
If you would create a common init instead, you wouldn't need register protection boolean. Have another function to set the callback.

jheiskan81 replied:
OK I create a common init.

nordicjm committed:
manifest: Remove civetweb from zephyr allow list

Civetweb was removed from zephyr therefore it no longer needs to be
part of the allow list

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

jheiskan81 commented:
@jarlamsa I fx you comment- I realize that I can register event handler multiple time it is not problem. link control code detect if handler is already register it not allocate new. I remove boolean because there was not any advantage.

jheiskan81 commented:
I close PR make PR to main branch.
